internal/service/streamprocessor/resource.go

```diff
@@ -0,0 +1,258 @@
+package streamprocessor
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+	"go.mongodb.org/atlas-sdk/v20240805001/admin"
+)
+
+const StreamProcessorName = "stream_processor"
+
+var _ resource.ResourceWithConfigure = &streamProcessorRS{}
+var _ resource.ResourceWithImportState = &streamProcessorRS{}
+
+func Resource() resource.Resource {
+	return &streamProcessorRS{
+		RSCommon: config.RSCommon{
+			ResourceName: StreamProcessorName,
+		},
+	}
+}
+
+type streamProcessorRS struct {
+	config.RSCommon
+}
+
+func (r *streamProcessorRS) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = ResourceSchema(ctx)
+}
+
+func (r *streamProcessorRS) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan TFStreamProcessorRSModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	streamProcessorReq, diags := NewStreamProcessorReq(ctx, &plan)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	var needsStarting bool
+	if !plan.State.IsNull() && !plan.State.IsUnknown() {
+		switch plan.State.ValueString() {
+		case StartedState:
+			needsStarting = true
+		case CreatedState:
+			needsStarting = false
+		default:
+			resp.Diagnostics.AddError("When creating a stream processor, the only valid states are CREATED and STARTED", "")
+		}
+	}
+
+	connV2 := r.Client.AtlasV2
+	projectID := plan.ProjectID.ValueString()
+	instanceName := plan.InstanceName.ValueString()
+	processorName := plan.ProcessorName.ValueString()
+	_, _, err := connV2.StreamsApi.CreateStreamProcessor(ctx, projectID, instanceName, streamProcessorReq).Execute()
+
+	if err != nil {
+		resp.Diagnostics.AddError("error creating resource", err.Error())
+		return
+	}
+
+	streamProcessorParams := &admin.GetStreamProcessorApiParams{
+		GroupId:       projectID,
+		TenantName:    instanceName,
+		ProcessorName: processorName,
+	}
+
+	streamProcessorResp, err := WaitStateTransition(ctx, streamProcessorParams, connV2.StreamsApi, []string{InitiatingState, CreatingState}, []string{CreatedState, FailedState})
```

Comment:
[Same comment](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2501#discussion_r1722983368)

Reply:
Removed in [1e09f4e](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2501/commits/1e09f4e3d4f54ad618499d2a7a71cb8c1237533c)